src/iris/api.py

```diff
@@ -1204,6 +1233,24 @@ def on_post(self, req, resp):
 
                 if not app:
                     raise HTTPBadRequest('Invalid application')
+            if 'dynamic_targets' in incident_params:
+                num_targets = session.execute('SELECT COUNT(DISTINCT `dynamic_index`) FROM `plan_notification` '
+                                              'WHERE `plan_id` = :plan_id',
+                                              {'plan_id': plan_id}).scalar()
+                if num_targets != len(incident_params['dynamic_targets']):
+                    raise HTTPBadRequest('Invalid number of dynamic targets')
+
+                for dynamic_target in incident_params.get('dynamic_targets', []):
+                    target = session.execute('''SELECT `target_role`.`id` AS `role_id`, `target`.`id` AS `target_id`
```

Comment:
i think this could be shortened to `session.execute().scalar()`, which would be None if there was no rows returned, then you can check that instead of rowcount